src/Reopt/TypeInference/ConstraintGen.hs

```diff
@@ -363,39 +370,39 @@ maybeGlobalTC ty soff = do
 -- freshRowVar :: CGenM ctx arch RowVar
 -- freshRowVar = inSolverM S.freshRowVar
 
-emitEq :: Ty -> Ty -> CGenM ctx arch ()
-emitEq t1 t2 = inSolverM (eqTC t1 t2)
+emitEq :: ConstraintProvenance -> Ty -> Ty -> CGenM ctx arch ()
```

Comment:
That is an intriguing idea. I'm not sure if it would be applicable in all places that we pass around constraints, however. For instance, I'm not sure what constraint we'd use for `SolverM`, which doesn't have anything like `CGenM`'s `ctx` type parameter.

I'll open an issue to track this idea.

Reply:
#249